distributed/protocol/pickle.py

```diff
@@ -1,19 +1,34 @@
 from __future__ import annotations
 
 import inspect
+import io
 import logging
 import pickle
 
 import cloudpickle
 from packaging.version import parse as parse_version
 
+from distributed.protocol.serialize import dask_deserialize, dask_serialize
+
 CLOUDPICKLE_GTE_20 = parse_version(cloudpickle.__version__) >= parse_version("2.0.0")
 
 HIGHEST_PROTOCOL = pickle.HIGHEST_PROTOCOL
 
 logger = logging.getLogger(__name__)
 
 
+class _DaskPickler(pickle.Pickler):
+    def reducer_override(self, obj):
+        if _always_use_pickle_for(obj):
+            return NotImplemented
```

Comment:
This is an attempt to deal with the segfault. returning `NotImplemented` will just fall back to default pickle.

The case that segfaults is actually a nested numpy array. I don't know how/why this would matter but maybe somebody else has more insight